### PR TITLE
feat(instance): new datasource alicloud_instance_type_families.

### PR DESCRIPTION
**Pull request by Lexsss**

### CODE REVIEW

File: website/docs/d/instance_type_families.html.markdown [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `zone_id` - (Optional, ForceNew) The Zone to launch the instance.
+* `generation` - (Optional) The generation of the instance type family. For more information, see [Instance type families](https://www.alibabacloud.com/help/doc-detail/25378.htm). 
+* `instance_charge_type` - (Optional, ForceNew) Valid values are `PrePaid`, `PostPaid`, Default to `PostPaid`.
+* `spot_strategy` - (Optional, ForceNew) Filter the results by ECS spot type. Valid values: `NoSpot`, `SpotWithPriceLimit` and `SpotAsPriceGo`. Default to `NoSpot`.
+* `output_file` - (Optional) File name where to save data source results (after running `terraform plan`).
```
Comment (YY-tomorrow) [inline website/docs/d/instance_type_families.html.markdown]: ids

File: website/docs/d/instance_type_families.html.markdown [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+layout: "alicloud"
+page_title: "Alicloud: alicloud_instance_type_families"
+sidebar_current: "docs-alicloud-datasource-instance-type-families"
+description: |-
+    Provides a list of ECS Instance Type Families to be used by the alicloud_instance resource.
+---
+
+# alicloud\_instance\_type\_families
+
+This data source provides the ECS instance type families of Alibaba Cloud.
```
Comment (YY-tomorrow) [inline website/docs/d/instance_type_families.html.markdown]: add note Available in 1.53.0+

File: website/docs/d/instance_type_families.html.markdown (hunk 1 above)
Comment (Lexsss) [reply website/docs/d/instance_type_families.html.markdown]: maybe no need

File: alicloud/data_source_alicloud_instance_type_families.go [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+func dataSourceAlicloudInstanceTypeFamilies() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAlicloudInstanceTypeFamiliesRead,
+
+		Schema: map[string]*schema.Schema{
+			"generation": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
```
Comment (xiaozhu36) [inline alicloud/data_source_alicloud_instance_type_families.go]: validation

File: website/docs/d/instance_type_families.html.markdown [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+-> **NOTE:** Available in 1.53.0+
```
Comment (xiaozhu36) [inline website/docs/d/instance_type_families.html.markdown]: 1.54.0

**Pull request merged**